### PR TITLE
Route upload

### PR DESCRIPTION
**Pull request by TheJulienM**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -8,5 +11,31 @@ def hello_world():  # put application's code here
     return render_template("index.html")
 
 
+@app.route('/parse-csv', methods=['POST', 'GET'])
+def parse_csv():
+    message = None
+    if request.method == 'POST':
```
Comment (BullShitProd) [inline app.py:17]: Use a const or variable for that

File: app.py [hunk 2]
```diff
@@ -8,5 +11,31 @@ def hello_world():  # put application's code here
...
+        print(request.files)
```
Comment (BullShitProd) [inline app.py:18]: Delete print

File: app.py [hunk 3]
```diff
@@ -8,5 +11,31 @@ def hello_world():  # put application's code here
...
+        file = request.files['upload']
```
Comment (BullShitProd) [inline app.py:19]: After that, you can create a new function on a service to upload the file.

File: app.py [hunk 4]
```diff
@@ -8,5 +11,31 @@ def hello_world():  # put application's code here
...
+        original_filename = os.path.splitext(file.filename)[0]
+        random_filename = secure_filename(original_filename) + "-" + uuid.uuid4().hex + ".csv"
+        file.save(os.path.join('upload', random_filename))
+        message = original_filename
+        # Nicolas fait de la magie avec pandas
+        # Affichage des résultats + Téléchargement du fichier csv pandaifié
+    elif request.method == 'GET':
```
Comment (BullShitProd) [inline app.py:26]: Same use const/variable

File: app.py [hunk 5]
```diff
@@ -8,5 +11,31 @@ def hello_world():  # put application's code here
...
+        return redirect("/", code=302)
```
Comment (BullShitProd) [inline app.py:27]: Same for the code

File: app.py [hunk 6]
```diff
@@ -8,5 +11,31 @@ def hello_world():  # put application's code here
...
+
+    response = {
+        'message': message,
+        'status': 200
```
Comment (BullShitProd) [inline app.py:31]: Same

File: static/script.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+document.querySelector('#upload')
+    .addEventListener('change', event => {
+        showSpinner();
+        setTimeout(function() {
+            document.querySelector('#formUpload').submit();
+        }, 1000);
```
Comment (BullShitProd) [inline static/script.js:6]: Use  const : 
``` 
const SPINNER_DURATION_MS = 1000
```

File: templates/index.html [hunk 8]
```diff
@@ -2,20 +2,26 @@
...
-        <div class="outer_quare">
-            <label id="inner_square" class="pointer inner_square" for="upload">
-                <div id="loader" class="hide lds-ring"><div></div><div></div><div></div><div></div></div>
-                <input id="upload" class="upload" type="file" accept=".csv" onchange=switchStep() />
-                <img id="csv_icon" class="csv_icon" src="csv.svg" alt="svg file logo" />
-                <span id="label" class="label">Click here or drop your csv file</span>
-            </label>
-        </div>
+        <form id="formUpload" method="POST" action="http://localhost:5000/parse-csv" enctype="multipart/form-data">
+            <div class="outer_quare">
```
Comment (BullShitProd) [inline templates/index.html:14]: action="parse-csv" (or like stuff) don't work ?

Comment (TheJulienM) [reply templates/index.html:14]: It's totally possible !

File: app.py [hunk 9]
```diff
@@ -1,12 +1,61 @@
...
+    "GET": "GET"
+}
+
+HTTP_CODE = {
+    "OK": 200,
+    "FOUND": 302
+}
 
 app = Flask(__name__)
+# app.config['UPLOAD_FOLDER'] = '/path/o/the/uy'
```
Comment (aymerickmichelet) [inline app.py:16]: delete comment

File: upload/.gitignore [hunk 10]
```diff
@@ -0,0 +1,3 @@
+*
+*/
+!.gitignore
```
Comment (aymerickmichelet) [inline upload/.gitignore:3]: add newline at end of file

**Review by BullShitProd — APPROVED** [review]

**Review by aymerickmichelet — APPROVED** [review]

**Pull request merged**